showcase-app/plugin/identityPlugin.js

```diff
@@ -44,28 +58,40 @@ export default class IdentityWidget {
 
     if (message.type === 'active-identity-response') {
       if (message.data) {
+        // All identities either have an IDP of WebID stored onto them
+        // We want to determine which one to use
         let idpOrWebID = message.data.idp;
 
         if (message.data.webID) {
+          // IF the identity has a WebID, we need to figure out which IDP to redirect to
           const idps = await this._getIDPsFromWebID(message.data.webID)
 
+          // In the end we set the IDP anyway. User gets redirected to IDP. If it supports more than one WebID, the user can confirm or select the correct one there.
           if (idps.length) {
             idpOrWebID = idps[0];
           }
         }
 
+        // Make the received identity actual
         this.activeIdentity = {
           ...message.data,
           idpOrWebID,
         };
 
+        // Fire the change handler
+        // The Solid App can subscribe to this event to handle any changes in the app to triggere required behaviour
         this.identityChangedHandler({
           ...this.activeIdentity,
         });
       }
     }
   }
 
+  /**
+   * Queries for an IDP for a given WebID
+   * We use the IDPs generally for redirecting the user to the login/authorization flow of their IDP
+   * TODO: If your WebID does not exist or the IDP cannot be determined, this will fail with an error and no fallback
+   */
   async _getIDPsFromWebID(webId) {
```

Comment:
I think this should happen by the extension. That way the extension always returns the IDP. This is not security concern, because this data is public in any case and there is not authentication happening at this point.